packages/forms/resources/views/components/textarea.blade.php

```diff
@@ -13,9 +13,10 @@
             ax-load-src="{{ \Filament\Support\Facades\FilamentAsset::getAlpineComponentSrc('textarea', 'filament/forms') }}"
             x-data="textareaFormComponent({ initialHeight: @js($initialHeight) })"
             x-ignore
-            x-on:load.window="render()"
+            x-init="render()"
             x-on:input="render()"
             x-on:resize.window="render()"
+            wire:ignore
```

Comment:
Doesn't this have negative side effects?
I think this breaks error state?

Reply:
@qrasa1 suggested this. I am not aware of what the side effects are

Reply:
I mean... it shouldn't? I'm not a Livewire expert by any means, but if it did have some negative side effects, wouldn't that be mentioned in the [docs](https://laravel-livewire.com/docs/2.x/alpine-js#ignoring-dom-changes)?

Reply:
It will break conditionally disabling the field, we can't use wire:ignore here

Reply:
Also the error styling wouldn't be rendered.

Reply:
I see... here is one possible workaround:
`textarea.blade.php` (note the `h-full` near the end):
```blade
    <div
        @if ($shouldAutosize = $shouldAutosize())
            ax-load
            ax-load-src="{{ \Filament\Support\Facades\FilamentAsset::getAlpineComponentSrc('textarea', 'filament/forms') }}"
            x-data="textareaFormComponent({ initialHeight: @js($initialHeight) })"
            x-ignore
            wire:ignore.self
            x-on:input="render()"
            x-on:resize.window="render()"
            style="height: {{ $initialHeight }}rem"
        @endif
    >
        <textarea
            @if($shouldAutosize)
                {{ $getExtraAlpineAttributeBag() }}
            @endif
            {{
                $attributes
                    ->merge([
                        'autocomplete' => $getAutocomplete(),
                        'autofocus' => $isAutofocused(),
                        'cols' => $getCols(),
                        'disabled' => $isDisabled(),
                        'id' => $getId(),
                        'maxlength' => (! $isConcealed) ? $getMaxLength() : null,
                        'minlength' => (! $isConcealed) ? $getMinLength() : null,
                        'placeholder' => $getPlaceholder(),
                        'readonly' => $isReadOnly(),
                        'required' => $isRequired() && (! $isConcealed),
                        'rows' => $rows,
                        $applyStateBindingModifiers('wire:model') => $statePath,
                    ], escape: false)
                    ->merge($getExtraAttributes(), escape: false)
                    ->merge($getExtraInputAttributes(), escape: false)
                    ->class([
                        'fi-fo-textarea block w-full h-full rounded-lg border-none bg-white px-3 py-1.5 text-base text-gray-950 shadow-sm outline-none ring-1 transition duration-75 placeholder:text-gray-400 focus:ring-2 disabled:bg-gray-50 disabled:text-gray-500 disabled:[-webkit-text-fill-color:theme(colors.gray.500)] disabled:placeholder:[-webkit-text-fill-color:theme(colors.gray.400)] dark:bg-white/5 dark:text-white dark:placeholder:text-gray-500 dark:disabled:bg-transparent dark:disabled:text-gray-400 dark:disabled:[-webkit-text-fill-color:theme(colors.gray.400)] dark:disabled:placeholder:[-webkit-text-fill-color:theme(colors.gray.500)] sm:text-sm sm:leading-6',
                        'ring-gray-950/10 focus:ring-primary-600 dark:ring-white/20 dark:focus:ring-primary-500 dark:disabled:ring-white/10' => ! $errors->has($statePath),
                        'ring-danger-600 focus:ring-danger-600 dark:ring-danger-500 dark:focus:ring-danger-500' => $errors->has($statePath),
                    ])
            }}
        ></textarea>
    </div>
``` 
`textarea.js`:
```javascript
export default function textareaFormComponent({ initialHeight }) {
    return {
        init: function () {
          setTimeout(() => this.render())
        },
        render: function () {
            if (this.$el.children[0].scrollHeight > 0) {
                this.$el.style.height = initialHeight + 'rem'
                this.$el.style.height = this.$el.children[0].scrollHeight + 'px'
            }
        },
    }
}
``` 
I thought that the `$getExtraAlpineAttributeBag()` belongs to the text area so I added there.

Reply:
Ok thank you for the explanation of the drawbacks. Is there any event we can listen for to call the `render()` when you save the edit form?

Reply:
One solution I have found is to add `wire:key="{{ rand() }}"` which forces Livewire to destroy the old components and to create a new one doing again the `init()`

Reply:
We should not do that, its bad practice to use a random key